subctl/diagnore-pre-install.md

```diff
@@ -0,0 +1,102 @@
+# Subctl `diagnose pre-install`
+
+## Summary
+
+As a user of Submariner, it would be useful in a myriad of situations to have a tool capable of testing the Kubernetes infrastructure
+where Submariner is to be installed.
+The tool should check all of Submariner's pre-requisites from the infrastructure, and report any deficiencies.
+It would be nice if the tool can also offer guidance for the deficiencies, so that the user can address them and install Submariner successfully.
+
+This tool would compliment other `subctl diagnose` commands which check that Submariner is installed and working properly, by allowing
+similar checks **before** the actual installation.
+
+Alternatively, the new command could be hosted elsewhere, if nesting under `diagnose` is deemed confusing.
+
+## Proposal
+
+Add a `subctl diagnose pre-install` command which checks that the infrastructure supports installing Submariner by checking:
+
+* Gateway nodes are set-up.
+  * Kube proxy, if there, doesn't run in IPVS mode - as it's currently not supported.
+* The broker is reachable from the cluster.
+* K8s version is supported.
+* The CNI is supported.
+* Conflicts with existing clusters in the clusterset.
+  * CIDRs should be distinct, unless the clusterset is using globalnet.
+* The necessary ports are open.
+  * On the gateway, when not using Load Balancer, the cable driver port(s).
```

Comment:
Again, at the point where the tool is going to be used Submariner is not deployed yet. How can we tell if the user intends to use `LoadBalancer` or not? Or are you suggesting that we assume that ROSA and ARO must use it, and all other platforms not?

Reply:
We could detect it for some platforms, but for starters we can get the info from the user.

I'll add some clarification